packages/vm/src/backends/singlepass.rs

```diff
@@ -43,13 +44,75 @@ pub fn backend() -> &'static str {
 
 /// Set the amount of gas units that can be used in the instance.
 pub fn set_gas_limit(instance: &mut WasmerInstance, limit: u64) {
-    let used = GAS_LIMIT.saturating_sub(limit);
-    metering::set_points_used(instance, used)
+    if limit > MAX_GAS_LIMIT {
+        panic!(
+            "Attempted to set gas limit larger than max gas limit (got: {}; maximum: {}).",
+            limit, MAX_GAS_LIMIT
+        );
+    } else {
```

Comment:
will this error be seen in the transaction on-chain later?

Reply:
I'm not sure what happens when the VM panics. But I assume this to force-halt the chain, which is the desired behaviour in this case, as it is a misconfiguration of the VM.

Reply:
i agree that it's unrealistic, because it is an extremely high limit, but this theoretically could fail. It won't crash the chain though, just fail the tx IIRC. I think this message won't be displayed, but i wanted to see if you remember/know otherwise.